### ISSUE TITLE
Missing block: Displays undefined block name for some unsupported blocks

### ISSUE THREAD

**Issue by fluiddot** [WordPress/gutenberg#31545]
## Description

The missing block is not showing the proper name of all unsupported blocks.

I checked the recently merged commits and looks like this bug was introduced in this [PR](https://github.com/WordPress/gutenberg/pull/31120).

**NOTE:** This issue happens on both platforms (iOS + Android).

## Step-by-step reproduction instructions

1. Add some unsupported blocks
2. Add a Table block, which is also unsupported (use the example below)
```
<!-- wp:table -->
<figure class="wp-block-table"><table><tbody><tr><td>1</td><td>2</td></tr><tr><td>3</td><td>4</td></tr></tbody></table></figure>
<!-- /wp:table -->
```
3. Open the post/page in the mobile app
4. Observe that for the table block is not displaying the block's name

## Expected behaviour

The missing block should display the block's name of the unsupported block.

## Actual behaviour

The missing block is not displaying the block's name of all unsupported blocks.

## Screenshots or screen recording (optional)

Table block|Bottom-sheet displayed after tapping
--|--
<img src=https://user-images.githubusercontent.com/14905380/117310881-fdfb8480-ae83-11eb-9730-138cfd7da4ea.png width=500>|<img src=https://user-images.githubusercontent.com/14905380/117309634-e53e9f00-ae82-11eb-964d-24026b137259.png width=500>

## WordPress information
- WordPress version: N/A
- Gutenberg version: N/A
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: iPhone 8 (simulator)
- Operating system: iOS 14.2
- WordPress app version: `develop`

**Comment by hypest** [comment]
👋 @fluiddot , just to be sure, have you perhaps tried on Android too? I see iOS mentioned in the device info but not sure if it's iOS _only_. Thanks!

**Comment by mchowning** [comment]
I tested iOS 17.3.0.20210503 beta, and it doesn't show this issue, so I think this isn't a problem we need to address for the current release. I also wasn't able to reproduce the issue on Android's 17.2 release, so it seems like this may be a very recent issue.

**Comment by fluiddot** [comment]
>👋 @fluiddot , just to be sure, have you perhaps tried on Android too? I see iOS mentioned in the device info but not sure if it's iOS only. Thanks!

Yeah, this happens on Android too.

>I tested iOS 17.3.0.20210503 beta, and it doesn't show this issue, so I think this isn't a problem we need to address for the current release. I also wasn't able to reproduce the issue on Android's 17.2 release, so it seems like this may be a very recent issue.

Right, the [PR](https://github.com/WordPress/gutenberg/pull/31120) that introduced it was merged after the `1.52.0` code freeze so fortunately this won't be released to users, but we should fix it for the next version `1.53.0`.
